src/main/scala/is/hail/expr/ir/package.scala

```diff
@@ -35,4 +35,12 @@ package object ir {
           ir.GetField(s, f.name))
     }.toArray)
   }
+
+  def coerce[T](x: Code[_]): Code[T] = x.asInstanceOf[Code[T]]
+
+  def coerce[T](lr: LocalRef[_]): LocalRef[T] = lr.asInstanceOf[LocalRef[T]]
+
+  def coerce[T <: Type](x: Type): T = x.asInstanceOf[T]
```

Comment:
One of these is not like the other.  The others are no-ops because they are casts that assert erased types, while this one is an actual downcast.  You don't need to change it, just remarking.  Or you could rename it something like `downcast` (or even make it totally generic although the machinations to generalized `Type` here might not be worth it.